draft-packages/illustration/KaizenDraft/Illustration/Players/VideoPlayer.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useRef, useEffect } from "react"
+import { assetUrl } from "@kaizen/hosted-assets"
+import styles from "../style.module.scss"
+
+export type VideoPlayerProps = {
+  /**
+   * Specifies whether the animation plays as soon as it is rendered.
+   * If the user has enabled prefer-reduced-motion their preferences
+   * take precedent over this prop.
+   */
+  autoplay?: boolean
+
+  /**
+   * Replay from start when active animation reaches the end of the animation.
+   */
+  loop?: boolean
+
+  /**
+   * The path of the initial video asset to play. This is only used in
+   * situations where we have two clips - the `initialAnimation` will play
+   * once, then we will swap to the ambient animation and loop indefinitely
+   */
+  initialAnimation?: string
+
+  /**
+   * Fallback image. Used when rendering of the asset fails, or as a
+   * poster for the video player.
+   */
+  fallback: string
+
+  /**
+   * The path of the ambient animation. An ambient animation can be used without
+   * an initial animation.
+   */
+  ambientAnimation: string
+}
+
+/**
+ * TODO
+ * ambient and initial animation
+ * convert to mp4 for safari
+ */
+export const VideoPlayer = ({
+  autoplay = true,
+  loop = false,
+  fallback,
+  ambientAnimation,
+}: VideoPlayerProps) => {
+  const videoRef = useRef<HTMLVideoElement>(null)
+  const reducedMotionQuery = window.matchMedia(
+    "(prefers-reduced-motion: reduce)"
+  )
+  const [
+    prefersReducedMotion,
+    setPrefersReducedMotion,
+  ] = React.useState<boolean>(reducedMotionQuery.matches || false)
+
+  useEffect(() => {
+    /**
+     * Setting `muted` on the player is required in Safari/Webkit and older
+     * versions of Chome for the video to autoplay (regardless of whether
+     * the format contains an audio stream or not).
+     *
+     * React does not guarentee the `muted` attribute is set on the
+     * `video` element. So on load we force set this attribute. See issue:
+     * https://github.com/facebook/react/issues/10389
+     */
+    const { current: videoElement } = videoRef
+    if (videoElement !== null) {
+      videoElement.setAttribute("muted", "")
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!reducedMotionQuery.addEventListener || !window) return
+    const updateMotionPreferences = () => {
+      const { matches = false } = window.matchMedia(
+        "(prefers-reduced-motion: reduce)"
+      )
```

Comment:
Is this done twice on purpose? (once on line 50 and again here)

Reply:
Yup - I do an initial query to get the users preferences then set up this listener in the (fairly uncommon) scenario of someone updating their motion preferences after this component has been rendered.